canned/mysql.go

```diff
@@ -0,0 +1,95 @@
+package canned
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/testcontainers/testcontainers-go/wait"
+
+	_ "github.com/go-sql-driver/mysql"
+	"github.com/pkg/errors"
+	testcontainers "github.com/testcontainers/testcontainers-go"
+)
+
+type MySQLContainerRequest struct {
+	testcontainers.GenericContainerRequest
+	RootPassword string
+	Username     string
+	Password     string
+	Database     string
+}
+
+type mysqlContainer struct {
+	Container testcontainers.Container
+	db        *sql.DB
+	req       MySQLContainerRequest
+	ctx       context.Context
+}
+
+func (c *mysqlContainer) GetDriver(username, password, database string) (*sql.DB, error) {
```

Comment:
Consider taking this from the request and fallbacking to the defaults of the image